cmd/kubeadm/app/cmd/init.go

```diff
@@ -0,0 +1,196 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/api"
+	kubemaster "k8s.io/kubernetes/cmd/kubeadm/app/master"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	netutil "k8s.io/kubernetes/pkg/util/net"
+)
+
+var (
+	initDoneMsgf = dedent.Dedent(`
+		Kubernetes master initialised successfully!
+
+		You can now join any number of machines by running the following on each node:
+
+		kubeadm join --token %s %s
+		`)
+)
+
+// NewCmdInit returns "kubeadm init" command.
+func NewCmdInit(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
+	advertiseAddrs := &[]string{}
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Run this on the first machine.",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunInit(out, cmd, args, s, advertiseAddrs)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.PersistentFlags().StringVar(
+		&s.Secrets.GivenToken, "token", "",
+		`(optional) Shared secret used to secure cluster bootstrap. If none is provided, one will be generated for you.`,
+	)
+	cmd.PersistentFlags().StringSliceVar(
+		advertiseAddrs, "api-advertise-addresses", []string{},
+		`(optional) The IP addresses to advertise, in case autodetection fails.`,
+	)
+	cmd.PersistentFlags().StringSliceVar(
+		&s.InitFlags.API.ExternalDNSNames, "api-external-dns-names", []string{},
+		`(optional) The DNS names to advertise, in case you have configured them yourself.`,
+	)
+	cmd.PersistentFlags().IPNetVar(
+		&s.InitFlags.Services.CIDR, "service-cidr", *kubeadmapi.DefaultServicesCIDR,
+		`(optional) Use alterantive range of IP address for service VIPs, defaults to `+
+			kubeadmapi.DefaultServicesCIDRString,
+	)
+	cmd.PersistentFlags().IPNetVar(
+		&s.InitFlags.PodNetwork.CIDR, "pod-network-cidr", net.IPNet{},
+		`(optional) Specify range of IP addresses for the pod network. If set, the control plane will automatically allocate CIDRs for every node.`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.Services.DNSDomain, "service-dns-domain", kubeadmapi.DefaultServiceDNSDomain,
+		`(optional) Use alternative domain for services, e.g. "myorg.internal"`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.CloudProvider, "cloud-provider", "",
+		`(optional) Enable a specific cloud provider features (external load-balancers, storage, etc), e.g. "gce"`,
+	)
+	cmd.PersistentFlags().BoolVar(
+		&s.InitFlags.Schedulable, "schedule-pods-here", false,
```

Comment:
@mikedanese I largely agree with you, was just trying to find a logical UX-minded justification to doing otherwise. I'm happy to remove this flag, knowing that it's only one `kubeclt` call away (I thought `kubectl taint` didn't make it into 1.4 and one had to do `kubectl edit nodes` or something).